pyvrp/cpp/search/Route.h

```diff
@@ -81,6 +81,24 @@ class Route
         NodeStats(TimeWindowSegment const &tws);
     };
 
+    // TODO template different cases?
+    class Proxy
```

Comment:
This object is returned by calls to `Route::at`, `Route::between`, `Route::after`, and `Route::before`. It can then be casted to the type most appropriate for the call site - currently, one of `TWS`, `Load`, or `Distance`. This happens implicitly, making the `Proxy` a completely transparent implementation detail.

Reply:
(I'll probably move this into its own header if we continue with this, because I need to template it for several special cases - and then it gets big fast)